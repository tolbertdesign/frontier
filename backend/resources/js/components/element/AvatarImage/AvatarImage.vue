<template>
  <img
    :alt="alt"
    :src="src"
    :class="[size, 'shadow border rounded-full']"
    @click="emit"
  >
</template>

<script>
export default {
  name: 'AvatarImage',
  status: 'prototype',
  release: '1.0.0',
  props: {
    alt: {
      type: String,
      default: '',
    },
    src: {
      type: String,
      default: '/v3-assets/dashboard/images/dashboard-avatar.svg',
    },
    shouldEmit: {
      type: Boolean,
      default: false,
    },
    small: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    size () {
      return this.small ? 'w-9 h-9' : 'w-16 h-16'
    },
  },
  methods: {
    emit () {
      if (this.shouldEmit) {
        this.$emit('open')
      }
    },
  },
}
</script>
