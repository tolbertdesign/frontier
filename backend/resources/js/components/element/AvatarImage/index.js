import AvatarImage from './AvatarImage.vue'

export default AvatarImage
