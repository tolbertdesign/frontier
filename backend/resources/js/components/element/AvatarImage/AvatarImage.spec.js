import { shallowMount, createLocalVue } from '@vue/test-utils'
import AvatarImage from './AvatarImage.vue'

const localVue = createLocalVue()

const alt = 'alt'
const src = 'src'
const shouldEmit = true
const small = true

describe('AvatarImage', () => {
  it('Snapshop test for small AvatarImage and check computed class for size', () => {
    const wrapper = shallowMount(AvatarImage, {
      localVue,
      propsData: {
        alt: alt,
        src: src,
        shouldEmit: shouldEmit,
        small: small,
      },
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.vm.size).toBe('w-9 h-9')
  })
  it('Snapshop test for not small AvatarImage and check computed class for size', () => {
    const wrapper = shallowMount(AvatarImage, {
      localVue,
      propsData: {
        alt: alt,
        src: src,
        shouldEmit: shouldEmit,
        small: false,
      },
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.vm.size).toBe('w-16 h-16')
  })
})
