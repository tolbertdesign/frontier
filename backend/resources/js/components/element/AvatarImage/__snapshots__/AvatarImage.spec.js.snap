// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarImage Snapshop test for not small AvatarImage and check computed class for size 1`] = `<img alt="alt" src="src" class="w-16 h-16 shadow border rounded-full">`;

exports[`AvatarImage Snapshop test for small AvatarImage and check computed class for size 1`] = `<img alt="alt" src="src" class="w-9 h-9 shadow border rounded-full">`;
